projects/npm-tools/packages/npm-scripts/README.md

```diff
@@ -165,6 +165,14 @@ liferay-npm-scripts theme build
 
 Runs the "build" task, providing it with the configuration it needs to find core dependencies such as the [`liferay-frontend-theme-styled` base theme files](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-theme/frontend-theme-styled/src/main/resources/META-INF/resources/_styled).
 
+### types
+
+```sh
+liferay-npm-scripts types
```

Comment:
Are you thinking we should run this before we start compiling modules as part of `ant all`, like we do `yarn install`?

Reply:
We could, but that's not what I was thinking. The idea is that the artifacts get updated as part of individual `build` in each module, and they produce changes that should be committed along with that module.

So we accompany that with an `ci:test:sf` check that ensures they really are up-do-date (ie. were committed), and this subcommand is dual purpose:

- A convenience that enables us to do a "global refresh" if somebody ever merges something that hasn't passed SF.
- A thing we can use as an implementation detail of the `ci:test:sf` check to report when things are stale.

Reply:
Ok, that's the part I was missing. That'd be good and won't impact build time, which is already off the charts 👍

Reply:
It's written down somewhere (Jira ticket? commit message? other PR?) but the whole idea of using project references is to enable fast builds, because each project only has to worry about its own little bit of the build and you avoid expensive global operations.